version-catalog/README.md

```diff
@@ -0,0 +1,216 @@
+      override val version = "0.0.8"                 // libs.versions.dummy.gradlePlugin
+      override val module = "$group:my-dummy-plugin" // libs.dummy.gradlePlugin
+      override val id = "my-dummy-plugin"            // libs.plugins.dummy
+   }
+
+   object Runtime : CatalogEntry() {
+
+      // When an entry does not override the version, it will try to fetch it
+      // from the closest parental entry, which has one. For example, in this case,
+      // all libraries within "Runtime" entry will have version = "1.0.0".
+
+      val win by lib("$group:runtime-win")     // libs.dummy.runtime.win
+      val mac by lib("$group:runtime-mac")     // libs.dummy.runtime.mac
+      val linux by lib("$group:runtime-linux") // libs.dummy.runtime.linux
+
+      object Bom : CatalogEntry() {
+         override val version = "2.0.0"           // libs.versions.dummy.runtime.bom
+         override val module = "$group:dummy-bom" // libs.dummy.runtime.bom
+      }
+   }
+
+   // It's also possible to declare an extra bundle by a property delegate.
+   // Just like with extra modules.
+
+   val runtime by bundle( // libs.bundles.dummy.runtime
+      Runtime.Bom,
+      Runtime.win,
+      Runtime.mac,
+      Runtime.linux,
+   )
+}
+```
+
+
+## Overriding of items shipped by `SpineVersionCatalog`
+
+Sometimes, it happens that a projects needs to override items, shipped by the catalog.
+In most cases, it is needed to override one or more versions.
+
+Currently, Gradle does not [provide](https://github.com/gradle/gradle/issues/20836) 
+a clear way to perform overwrite for already created catalogs. Thus, we use approach
+with creating a catalog directly in settings files. It preserves a possibility
+of a local override.
+
+Below are examples on how to override the items of `Dummy` dependency. Instead of
+applying `SpineVersionCatalog`, `DummyVersionCatalog` is used, as we are going to
+override `Dummy`'s items.
+
+### Overriding of versions
+
+In order to override a version you should declare it *before* applying the catalog.
+A version, declared first always wins. All subsequent declarations of the same version
+will be ignored by the builder.
+
+In total, `Dummy` declares three versions.
+
+Let's override them all:
+
+```kotlin
+dependencyResolutionManagement {
+    versionCatalogs {
+        create("libs") {
+           
+            version("dummy", "2.0.0")              // Dummy.version
+            version("dummy-gradlePlugin", "0.0.9") // Dummy.GradlePlugin.version
+            version("dummy-runtime-bom", "3.0.0")  // Dummy.Runtime.Bom.version
+           
+            DummyVersionCatalog.useIn(this)
+        }
+    }
+}
+```
+
+### Overriding of libraries, plugins and bundles
+
+In order to override a library, plugin or bundle, one should declare it *after* 
+applying the catalog. This is the opposite of what is done with versions.
+
+When overriding libraries and plugins, a version should be specified in-place.
+
+For example:
```

Comment:
In this example, there should be more comments which _literally point_ to the rules which you describe in the sentences above.